### PR TITLE
Add catch prop function

### PR DESCRIPTION
**Pull request by dimas-cyriaco**
This PR add a `catch` prop function that can be used to handle the promise rejection locally.

### CODE REVIEW

**Comment by jomaxx** [comment]
thanks for the PR!

File: src/Async.js [hunk 1]
```diff
@@ -77,6 +77,10 @@ class Async extends React.Component {
       case 1:
         return this.props.children(value);
       case 2:
+        if (this.props.catch) {
+          return this.props.catch(error);
```
Comment (jomaxx) [inline src/Async.js]: could we do this instead?

```js
if (this.props.catch) {
  return this.props.children(this.props.catch(error));
}
```

that way all of the render logic can be in one place

```js
<Async
  await={promise}
  catch={error => ({ error })}
>
  {result => (
    <div>
      {result.error && <h1>{result.error.message}</h1>}
    </div>
  )}
</Async>
```

**Comment by dimascyriaco** [comment]
Done.

**Comment by dimas-cyriaco** [comment]
@jomaxx done.

**Review by jomaxx — APPROVED** [review]

**Comment by jomaxx** [comment]
@dimascyriaco thanks for the contribution!

**Comment by jomaxx** [comment]
released in 1.4.0

**Pull request merged**